lib/finsta/posts.ex

```diff
@@ -62,36 +62,44 @@ defmodule Finsta.Posts do
 
   ## Examples
 
-      iex> update_post(post, %{field: new_value})
+      iex> update_post(user, post, %{field: new_value})
       {:ok, %Post{}}
 
-      iex> update_post(post, %{field: bad_value})
+      iex> update_post(user, post, %{field: bad_value})
       {:error, %Ecto.Changeset{}}
 
   """
-  def update_post(%Post{} = post, attrs) do
-    post
-    |> Post.changeset(attrs)
-    |> Repo.update()
-    |> tap(&broadcast_post(&1, :update))
+  def update_post(user, %Post{} = post, attrs) do
+    if post.user_id == user.id do
+      post
+      |> Post.changeset(attrs)
+      |> Repo.update()
+      |> tap(&broadcast_post(&1, :update))
+    else
+      {:error, :unauthorized}
+    end
   end
 
   @doc """
   Deletes a post.
 
   ## Examples
 
-      iex> delete_post(post)
+      iex> delete_post(user, post)
       {:ok, %Post{}}
 
-      iex> delete_post(post)
+      iex> delete_post(user, post)
       {:error, %Ecto.Changeset{}}
 
   """
-  def delete_post(%Post{} = post) do
-    post
-    |> Repo.delete()
-    |> tap(&broadcast_post(&1, :delete))
+  def delete_post(user, %Post{} = post) do
+    if post.user_id == user.id do
+      post
+      |> Repo.delete()
+      |> tap(&broadcast_post(&1, :delete))
+    else
+      {:error, :unauthorized}
```

Comment:
I agree with doing the authorization at this level (context) if you don't plan to show these errors to users. However, I would use the post id in the query itself. I feel it is better to say "record not found" when the post found is one from another user. In other words, using an post id from another user produces the same error than using an invalid or non-existing post id, which I think it is the intention.

Reply:
Makes a lot of sense 🙌